### PR TITLE
add device migration metrics

### PR DESCRIPTION
**Pull request by florezita**
Fulfilling this [ticket](https://mozilla-hub.atlassian.net/browse/DS-2015?atlOrigin=eyJpIjoiMWI1ZjMxOGY4NWFlNDY2Nzk5MmZhYmM2YmQ5OGQwODEiLCJwIjoiaiJ9). 

Used definitions from  https://sql.telemetry.mozilla.org/queries/86549/source.

### CODE REVIEW

File: definitions/firefox_desktop.toml [hunk 1]
```diff
@@ -858,6 +858,69 @@ description = """
 Was Firefox pinned to the Windows Taskbar at any point during the interval?
 """
 
+[metrics.imported_bookmarks]
+data_source = "main"
+select_expression = "COUNTIF(COALESCE((SELECT SUM(`mozfun`.hist.extract(x.value).sum) FROM UNNEST(payload.keyed_histograms.fx_migration_bookmarks_quantity) x WHERE x.key != "firefox"), 0) > 0)"
```
Comment (scholtzan) [inline definitions/firefox_desktop.toml:863]:
```suggestion
select_expression = "COUNTIF(COALESCE((SELECT SUM(`mozfun`.hist.extract(x.value).sum) FROM UNNEST(payload.keyed_histograms.fx_migration_bookmarks_quantity) x WHERE x.key != 'firefox'), 0) > 0)"
```

File: definitions/firefox_desktop.toml [hunk 2]
```diff
@@ -858,6 +858,69 @@ description = """
 Was Firefox pinned to the Windows Taskbar at any point during the interval?
 """
 
+[metrics.imported_bookmarks]
+data_source = "main"
+select_expression = "COUNTIF(COALESCE((SELECT SUM(`mozfun`.hist.extract(x.value).sum) FROM UNNEST(payload.keyed_histograms.fx_migration_bookmarks_quantity) x WHERE x.key != 'firefox'), 0) > 0)"
+type = "scalar"
+frienly_name = "Imported Bookmarks"
```
Comment (dmose) [inline definitions/firefox_desktop.toml:865]:
```suggestion
friendly_name = "Imported Bookmarks"
```

File: definitions/firefox_desktop.toml [hunk 3]
```diff
@@ -858,6 +858,69 @@ description = """
...
+description = "Count of clients who imported logins"
+category = "Device Migration"
+owner = "nflorez@mozilla.com"
+deprecated = false
+
+[metrics.imported_logins_count]
+data_source = "clients_daily"
+select_expression = "NULLIF(SUM(logins_migrations_quantity_all), 0)"
+type = "scalar"
+friendly_name = "Average Number of Imported Logins"
```
Comment (scholtzan) [inline definitions/firefox_desktop.toml:896]:
```suggestion
friendly_name = "Total Number of Imported Logins"
```

File: definitions/firefox_desktop.toml [hunk 4]
```diff
@@ -858,6 +858,69 @@ description = """
...
+category = "Device Migration"
+owner = "nflorez@mozilla.com"
+deprecated = false
+
+
+[metrics.imported_history_count]
+data_source = "clients_daily"
+select_expression = "NULLIF(SUM(history_migrations_quantity_all), 0)"
+type = "scalar"
+friendly_name = "Average Number of Imported History Links"
```
Comment (scholtzan) [inline definitions/firefox_desktop.toml:917]:
```suggestion
friendly_name = "Total Number of Imported History Links"
```

File: definitions/firefox_desktop.toml [hunk 5]
```diff
@@ -858,6 +858,69 @@ description = """
...
+category = "Device Migration"
+owner = "nflorez@mozilla.com"
+deprecated = false
+
+
+[metrics.imported_bookmarks_count]
+data_source = "clients_daily"
+select_expression = "NULLIF(SUM(bookmark_migrations_quantity_all), 0)"
+type = "scalar"
+friendly_name = "Average Number of Imported Bookmarks"
```
Comment (scholtzan) [inline definitions/firefox_desktop.toml:876]:
```suggestion
friendly_name = "Total Number of Imported Bookmarks"
```

File: jetstream/first-run-easy-as-one-click-ver-1.toml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+## Imported History
+[metrics.imported_history]
+data_source = "main"
+select_expression = "COUNTIF(COALESCE((SELECT SUM(`mozfun`.hist.extract(x.value).sum) FROM UNNEST(payload.keyed_histograms.fx_migration_history_quantity) x WHERE x.key != 'firefox'), 0) > 0)"
+type = "scalar"
+friendly_name = "Imported History"
+description = "Count of clients who imported history"
+category = "Device Migration"
+[metrics.imported_history.statistics.binomial]
+
```
Comment (scholtzan) [inline jetstream/first-run-easy-as-one-click-ver-1.toml:44]:
```suggestion
```

No need to redefine the metrics here, you can just reference their slug

File: jetstream/onboarding-welcome-video-experiment.toml [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+## Imported History
+[metrics.imported_history]
+data_source = "main"
+select_expression = "COUNTIF(COALESCE((SELECT SUM(`mozfun`.hist.extract(x.value).sum) FROM UNNEST(payload.keyed_histograms.fx_migration_history_quantity) x WHERE x.key != 'firefox'), 0) > 0)"
+type = "scalar"
+friendly_name = "Imported History"
+description = "Count of clients who imported history"
+category = "Device Migration"
+[metrics.imported_history.statistics.binomial]
```
Comment (scholtzan) [inline jetstream/onboarding-welcome-video-experiment.toml:43]:
```suggestion
```

File: jetstream/first-run-easy-as-one-click-ver-1.toml (hunk 6 above)
Comment (florezita) [reply jetstream/first-run-easy-as-one-click-ver-1.toml:44]: their slug is their name? for example, referencing in the metrics section by name should be enough?

```
[metrics]
weekly = ['is_pinned', 'imported_bookmarks', 'imported_logins', 'imported_history']
overall = ['is_pinned', 'imported_bookmarks', 'imported_logins', 'imported_history']
```

Comment (scholtzan) [reply jetstream/first-run-easy-as-one-click-ver-1.toml:44]: yes, that's how you'd reference them

**Comment by florezita** [comment]
Not sure why this is happening because I did edit `jetstream/definitions/firefox_desktop.toml` to incorporate how the metrics would be aggregated and it was passing before I edited the experiment files. An easy fix would be to include the relevant lines to the experiment files. Thoughts or suggestions?

File: jetstream/definitions/firefox_desktop.toml [hunk 8]
```diff
@@ -70,3 +70,14 @@ drop_highest = 0
...
+[metrics.import_bookmarks_count.statistics.bootstrap_mean]
+pre_treatments = ["remove_nulls"]
+
+[metrics.import_logins.statistics.binomial]
+[metrics.import_logins_count.statistics.bootstrap_mean]
+pre_treatments = ["remove_nulls"]
+
+[metrics.import_history.statistics.binomial]
+[metrics.import_history_count.statistics.bootstrap_mean]
+pre_treatments = ["remove_nulls"]
```
Comment (scholtzan) [inline jetstream/definitions/firefox_desktop.toml:83]:
```suggestion
[metrics.imported_bookmarks.statistics.binomial]
[metrics.imported_bookmarks_count.statistics.bootstrap_mean]
pre_treatments = ["remove_nulls"]

[metrics.imported_logins.statistics.binomial]
[metrics.imported_logins_count.statistics.bootstrap_mean]
pre_treatments = ["remove_nulls"]

[metrics.imported_history.statistics.binomial]
[metrics.imported_history_count.statistics.bootstrap_mean]
pre_treatments = ["remove_nulls"]
```

**Comment by scholtzan** [comment]
Just some typos in the metric names, I just pushed a fix

**Review by scholtzan — APPROVED** [review]

**Comment by florezita** [comment]
Thank you!

**Pull request merged**